clickhouse_connect/driver/asyncclient.py

```
@@ -22,13 +25,22 @@ class AsyncClient:
     Internally, each of the methods that uses IO is wrapped in a call to EventLoop.run_in_executor.
     """
 
-    def __init__(self, *, client: Client, executor_threads: int = 0):
+    def __init__(self,
+                 *,
+                 client: Client,
+                 executor_threads: int | None = None,
+                 executor: Union[Optional[ThreadPoolExecutor], object] = NEW_THREAD_POOL_EXECUTOR):
         if isinstance(client, HttpClient):
             client.headers['User-Agent'] = client.headers['User-Agent'].replace('mode:sync;', 'mode:async;')
         self.client = client
         if executor_threads == 0:
             executor_threads = min(32, (os.cpu_count() or 1) + 4)  # Mimic the default behavior
-        self.executor = ThreadPoolExecutor(max_workers=executor_threads)
+        if executor is NEW_THREAD_POOL_EXECUTOR:
+            self.new_executor = True
+            self.executor = ThreadPoolExecutor(max_workers=executor_threads)
+        else:
+            self.new_executor = False
+            self.executor = executor
```

We hit this branch if a user passed in their own executor. However, if they also supplied `executor_threads` it could be a little confusing as that gets silently ignored since the executor already brings along its own `max_executors` from when it was originally configured. Perhaps a simple debug logging statement `if executor_threads != 0:` add a note about something like "executor_threads is ignored when a custom executor is provided".